be/src/storage/rowset/dict_page.cpp

```diff
@@ -235,7 +235,9 @@ Status DictPageDecoder<Type>::next_batch(const SparseRange<>& range, Column* dst
         _dict_decoder->at_index(codewords[i], &value);
         numbers[i] = value;
     }
-    dst->append_numbers(numbers.data(), numbers.size() * SIZE_OF_TYPE);
+    if (dst->append_numbers(numbers.data(), numbers.size() * SIZE_OF_TYPE) == -1) {
```

Comment:
1. the result should be compared with numbers.size()
2. you need to assign a variable for return value.

Reply:
> 1. the result should be compared with numbers.size()
> 2. you need to assign a variable for return value.

fixed